### PR TITLE
Cleanup DTLog, implement log levels

### PR DESCRIPTION
**Pull request by MathNerd28**
- Add version suffix in DTLibInfo
- Remove duplicate mode logging
- Use tryFlush() instead of flush()
- Add constant LOG_PATH_SEPARATOR
- Use RobotController.getTeamNumber() instead of custom implementation
- Change unknown team number to 0
- General cleanup, suppress useless warnings

### CODE REVIEW

**Comment by MathNerd28** [comment]
Implement log levels for variables. They filter variables based on their immediate parent's log level, allowing components to be selectively logged based on the current level. Levels for variables are not denoted in the log output.

Also, properly implement log levels for messages. These are denoted in the log, under identifiers 0x0008 through 0x000B.

**Comment by MathNerd28** [comment]
Properly log exceptions where applicable (not during startup, DEBUG level when explicitly caught, etc.)

**Comment by MathNerd28** [comment]
This PR is ready for review

File: src/main/java/org/victorrobotics/dtlib/log/DTLogWriter.java [hunk 1]
```diff
@@ -63,44 +63,47 @@ public final class DTLogWriter implements Closeable, Flushable {
...
 
-    int team = getTeamNumber();
-    writeLong(((long) team << 48) | startTimeMillis);
+    long team = DTRobot.getTeamNumber();
+    writeLong((team << 48) | startTimeMillis);
     checksum ^= team << 16;
     checksum ^= (int) ((startTimeMillis >> 32) | startTimeMillis);
-
     writeInt(checksum);
+
```
Comment (mmarchetti) [inline src/main/java/org/victorrobotics/dtlib/log/DTLogWriter.java:106]: You might consider something like CRC32 for the checksum: https://docs.oracle.com/javase/8/docs/api/java/util/zip/CRC32.html

File: src/main/java/org/victorrobotics/dtlib/log/DTLogWriter.java [hunk 2]
```diff
@@ -368,77 +374,91 @@ public boolean logNewTimestamp() {
...
+      PrintStream debugPrinter = new PrintStream(debugOutput);
+      exception.printStackTrace(debugPrinter);
+      String str = new String(debugOutput.toByteArray());
+      return str.substring(str.indexOf('\t'));
+    });
+  }
 
+  public static void init(DTLog.Level logLevel) {
+    while (true) {
+      if (!RobotController.isSystemTimeValid()) {
```
Comment (mmarchetti) [inline src/main/java/org/victorrobotics/dtlib/log/DTLogWriter.java:448]: I'm not clear on the logic here. It seems like you want to wait until the system time IS valid, then create and return the instance. It seems like the check may be inverted.

**Review by mmarchetti — APPROVED** [review]

File: src/main/java/org/victorrobotics/dtlib/log/DTLogWriter.java (hunk 2 above)
Comment (MathNerd28) [reply src/main/java/org/victorrobotics/dtlib/log/DTLogWriter.java:448]: Copy-paste error, will fix in a moment

**Pull request merged**